File: codesLeftInWindows/CP-Problems-tilnoeneBackup/URI/2431.java
import java.util.Scanner;

public class Main {
    public static boolean dentro(int x, int y, int a, int b){
        if(x <= a && y <= b) return true;
        return false;
    }
    
    public static int min(int a, int b){
        if(a < b) return a;
        return b;
    }
    
    public static void main(String args[]){
        Scanner ler = new Scanner(System.in);
        
        int a1 = ler.nextInt(), b1 = ler.nextInt(), a2 = ler.nextInt(), b2 = ler.nextInt(), a = ler.nextInt(), b = ler.nextInt();
        boolean ans = false;
        int c1x = b1 + a2;
        int c1y = min(a1, b2);
        if(dentro(a, b, a1, b1) || dentro(b, a, a1, b1) || dentro(a, b, a2, b2) || dentro(b, a, a2, b2)){
        System.out.printf("S\n");
        ans = true;
    } else if(dentro(a, b, c1x, c1y) || dentro(b, a, c1x, c1y)){
        System.out.printf("S\n");
        ans = true;
    } else{
        c1x = a1 + b2;
        c1y = min(b1, a2);
        
        if(dentro(a, b, c1x, c1y) || dentro(b, a, c1x, c1y)){
            System.out.printf("S\n");
            ans = true;
        }
        
        c1x = a1 + a2;
        c1y = min(b1, b2);
        
        if(dentro(a, b, c1x, c1y) || dentro(b, a, c1x, c1y)){
            System.out.printf("S\n");
            ans = true;
        }
        
        c1x = b1 + b2;
        c1y = min(a1, a2);
        
        if(dentro(a, b, c1x, c1y) || dentro(b, a, c1x, c1y)){
            System.out.printf("S\n");
            ans = true;
        }
        
    }
    
    if(!ans) System.out.printf("N\n");
    }
}

File: codesLeftInWindows/CP-Problems-tilnoeneBackup/URI/2431.kt
import java.util.*
import java.lang.Math

fun dentro(x: Int, y: Int, a: Int, b: Int): Boolean {
    if(x <= a && y <= b) return true;
    return false;
}

fun min(a: Int, b: Int): Int {
    if(a < b) return a;
    return b;
}

fun main(args: Array<String>) {
    val ler = Scanner(System.`in`)

        var a1: Int = ler.nextInt();
        var b1: Int = ler.nextInt();
        var a2: Int = ler.nextInt();
        var b2: Int = ler.nextInt();
        var a: Int = ler.nextInt();
        var b: Int = ler.nextInt();
        var ans: Boolean = false;
        var c1x: Int = b1 + a2;
        var c1y: Int = min(a1, b2);
        if(dentro(a, b, a1, b1) || dentro(b, a, a1, b1) || dentro(a, b, a2, b2) || dentro(b, a, a2, b2)){
        println("S");
        ans = true;
    } else if(dentro(a, b, c1x, c1y) || dentro(b, a, c1x, c1y)){
        println("S");
        ans = true;
    } else{
        c1x = a1 + b2;
        c1y = min(b1, a2);
        
        if(dentro(a, b, c1x, c1y) || dentro(b, a, c1x, c1y)){
            println("S");
            ans = true;
        }
        
        c1x = a1 + a2;
        c1y = min(b1, b2);
        
        if(dentro(a, b, c1x, c1y) || dentro(b, a, c1x, c1y)){
            println("S");
            ans = true;
        }
        
        c1x = b1 + b2;
        c1y = min(a1, a2);
        
        if(dentro(a, b, c1x, c1y) || dentro(b, a, c1x, c1y)){
            println("S");
            ans = true;
        }
        
    }
    
    if(!ans) println("N");

}